OSDI22/microbench/test_parallel_shuffle.py

```diff
@@ -0,0 +1,102 @@
+import ray
+import csv
+import argparse
+import numpy as np
+import time
+from time import perf_counter
+
+####################
+## Argument Parse ##
+####################
+parser = argparse.ArgumentParser()
+parser.add_argument('--WORKING_SET_RATIO', '-w', type=int, default=4)
+parser.add_argument('--OBJECT_STORE_SIZE', '-o', type=int, default=1_005_000_000)
+parser.add_argument('--OBJECT_SIZE', '-os', type=int, default=40_000_000)
+parser.add_argument('--RESULT_PATH', '-r', type=str, default="../data/parallel_shuffle.csv")
+parser.add_argument('--NUM_TRIAL', '-t', type=int, default=5)
+args = parser.parse_args()
+params = vars(args)
+
+OBJECT_STORE_SIZE = params['OBJECT_STORE_SIZE'] 
+OBJECT_SIZE = params['OBJECT_SIZE'] 
+WORKING_SET_RATIO = params['WORKING_SET_RATIO']
+RESULT_PATH = params['RESULT_PATH']
+NUM_TRIAL = params['NUM_TRIAL']
+OBJECT_STORE_BUFFER_SIZE = 5_000_000 #this value is to add some space in ObjS for nprand metadata and ray object metadata
+
+def test_ray_shuffle():
+    shuffle_start = perf_counter()
+
+    @ray.remote
+    def map(npartitions):
+        data = np.random.rand(OBJECT_SIZE // 8)
+        size = OBJECT_SIZE//npartitions
+        return tuple(data[(i*size):((i+1)*size)] for i in range(npartitions))
+
+    @ray.remote
+    def reduce(*partitions):
+        return True
+
+    npartitions = OBJECT_STORE_SIZE//OBJECT_SIZE 
+    refs = []
+    for i in range(WORKING_SET_RATIO):
+        refs.append([map.options(num_returns=npartitions).remote(npartitions)
+            for _ in range(npartitions)])
+    results = []
+    for i in range(WORKING_SET_RATIO):
```

Comment:
I think the map tasks should also be submitted in this loop, right? Otherwise, we are actually only producing working set=1 amount of memory.